indigo/docs/guides/howto-hello-indigo-tyrian.md

```diff
@@ -0,0 +1,445 @@
+does this job instead.
+
+You'll want to upgrade `hello-indigo` to use `IndigoGame`. To do this either
+follow [this guide](howto-indigo-game.md)
+or replace `HelloIndigo.scala` with
+[this](https://gist.github.com/hobnob/c24f00936e91a7b7e5d644d19e4f1b32)
+
+Although Indigo builds and exports to Electron natively, for this project
+we'll export directly to HTML and use Yarn to run our web server, with
+ParcelJS to copy and package up our HTML, JS, and CSS.
+
+You can use NPM to install Yarn with the command `npm install yarn`, and then
+install ParcelJS with `yarn add parcel --dev`.
+
+### Setting up the Build
+
+Next you'll need to update either your
+`build.sc` (if using mill) to  with this:
+
+```diff
+   def buildGame() = T.command {
+     T {
+       compile()
+       fastOpt()
+-      indigoBuild()()
+     }
+   }
+
+-  def runGame() = T.command {
+-    T {
+-      compile()
+-      fastOpt()
+-      indigoRun()()
+-    }
+-  }
+-
+   val indigoVersion = "0.12.0"
++  val tyrianVersion = "0.3.0"
+
+   def ivyDeps =
+    Agg(
++     ivy"io.indigoengine::tyrian::$tyrianVersion",
++     ivy"io.indigoengine::tyrian-indigo-bridge::$tyrianVersion",
+      ivy"io.indigoengine::indigo-json-circe::$indigoVersion",
+      ivy"io.indigoengine::indigo::$indigoVersion",
+      ivy"io.indigoengine::indigo-extras::$indigoVersion"
+    )
+```
+
+ or `build.sbt` (if using sbt) to this:
+
+ ```diff
+ ```
+
+What we've done here is add Tyrian and the Tyrian Indigo Bridge to our build.
+Tyrian will deal with all of our HTML and the Indigo Bridge will deal with
+communication between Tyrian and Indigo. We've also removed `runGame` as this
+would usually run Electron.
+
+### Setting up the HTML and ParcelJS
+
+Usually Indigo will generate all of our HTML for us and run it through Electron,
+but for this example we're going to generate our own HTML so that we can
+inject CSS.
+
+Firstly we'll create an `app.js` file that simply loads Tyrian and then launches
+it for our page:
+
+```js
+import {
+    TyrianApp
+} from './out/HelloIndigo/fastOpt.dest/out.js';
+
+TyrianApp.launch("main");
+```
+
+We'll also need an `index.html` to hold our basic HTML data:
+
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+      <meta charset="UTF-8">
+      <title>Hello Indigo</title>
+      <link rel="stylesheet" href="css/main.css" />
+      <script type="module" defer src="app.js"></script>
+  </head>
+  <body>
+      <div id="main"></div>
+  </body>
+</html>
+```
+
+You'll notice we're using the direct JS output from our build here, which may
+feel odd. What will happen when we run ParcelJS through Yarn is that the HTML
+will be copied to a build directory along with the JS, CSS and any dependant
+static files (such as images) that may be needed.
+
+Now generate an empty `css/main.css`.
+
+Next we'll need to update the `package.json` so that ParcelJS will run when yarn
+starts and copies the static files for Indigo to use. To do this add the
+following to the top of `package.json`:
```

Comment:
What is the purpose of ParcelJS?

Is it only to copy static files from one place to another? At least in sbt this would idiomatically be done as part of the build, with a call to `sbt.IO.copyDirectory` in a task.

Reply:
In most of the Tyrian examples, Parcel.js is used as our bundler of choice (over something like Webpack). Superficially here we're just using it to do the asset copy true, but the expectation is that we're going to use it in the development of our site, linked up resources, bundling up a deployable site, running a dev-server with hot-reloading etc.
In the server-based examples we use scalajs-bundler instead, because the Http4s service build plays nicer with that bundler in that scenario.